flink-runtime/src/main/java/org/apache/flink/streaming/api/graph/AdaptiveGraphManager.java

```diff
@@ -0,0 +1,1053 @@
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+import static org.apache.flink.streaming.api.graph.StreamingJobGraphGenerator.addVertexIndexPrefixInVertexName;
+import static org.apache.flink.streaming.api.graph.StreamingJobGraphGenerator.configureCheckpointing;
+import static org.apache.flink.streaming.api.graph.StreamingJobGraphGenerator.connect;
+import static org.apache.flink.streaming.api.graph.StreamingJobGraphGenerator.createChainedMinResources;
+import static org.apache.flink.streaming.api.graph.StreamingJobGraphGenerator.createChainedName;
+import static org.apache.flink.streaming.api.graph.StreamingJobGraphGenerator.createChainedPreferredResources;
+import static org.apache.flink.streaming.api.graph.StreamingJobGraphGenerator.isChainable;
+import static org.apache.flink.streaming.api.graph.StreamingJobGraphGenerator.isChainableInput;
+import static org.apache.flink.streaming.api.graph.StreamingJobGraphGenerator.markSupportingConcurrentExecutionAttempts;
+import static org.apache.flink.streaming.api.graph.StreamingJobGraphGenerator.preValidate;
+import static org.apache.flink.streaming.api.graph.StreamingJobGraphGenerator.serializeOperatorCoordinatorsAndStreamConfig;
+import static org.apache.flink.streaming.api.graph.StreamingJobGraphGenerator.setAllOperatorNonChainedOutputsConfigs;
+import static org.apache.flink.streaming.api.graph.StreamingJobGraphGenerator.setManagedMemoryFraction;
+import static org.apache.flink.streaming.api.graph.StreamingJobGraphGenerator.setOperatorChainedOutputsConfig;
+import static org.apache.flink.streaming.api.graph.StreamingJobGraphGenerator.setOperatorConfig;
+import static org.apache.flink.streaming.api.graph.StreamingJobGraphGenerator.setPhysicalEdges;
+import static org.apache.flink.streaming.api.graph.StreamingJobGraphGenerator.setSlotSharingAndCoLocation;
+import static org.apache.flink.streaming.api.graph.StreamingJobGraphGenerator.setVertexDescription;
+import static org.apache.flink.streaming.api.graph.StreamingJobGraphGenerator.tryConvertPartitionerForDynamicGraph;
+import static org.apache.flink.streaming.api.graph.StreamingJobGraphGenerator.validateHybridShuffleExecuteInBatchMode;
+
+/** Default implementation for {@link AdaptiveGraphGenerator}. */
+@Internal
+public class AdaptiveGraphManager implements AdaptiveGraphGenerator {
```

Comment:
When creating a `JobGraph` through this class, the `UserArtifact`, `UserJarBlobKeys`, and `Classpaths` from the `StreamGraph` are lost. 

Additionally, these cases lack test coverage.